Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -25,6 +26,20 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
             // profile.bookmarks.modelForRoot(self.onNewModel, failure: self.onModelFailure)
         }
     }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "firefoxAccountChanged:", name: NotificationFirefoxAccountChanged, object: nil)
```

Comment:
Since we're deregistering in the deinit call, we should move the registration to the init call instead of viewDidLoad because there can be a chance that viewDidLoad doesn't get called.